File: test/proto/ts/smoke.ts
// Smoke check (not a test): prove the provider loads and normalizes.
// Run: node --experimental-strip-types test/proto/ts/smoke.ts
import { TestProvider, equal, equalStrict, errorMatches, structMatch } from './provider.ts'

const p = TestProvider.load()
const fns = p.functions()
console.log('functions:', fns.join(', '))

let total = 0
const kinds: Record<string, number> = {}
const inkinds: Record<string, number> = {}
for (const fn of fns) {
  for (const e of p.entries(fn)) {
    total++
    kinds[e.expect.kind] = (kinds[e.expect.kind] ?? 0) + 1
    inkinds[e.input.kind] = (inkinds[e.input.kind] ?? 0) + 1
  }
}
console.log('total entries:', total)
console.log('expect kinds:', JSON.stringify(kinds))
console.log('input kinds:', JSON.stringify(inkinds))

const first = p.entries('getpath', 'basic')[0]
console.log('sample getpath/basic[0]:', JSON.stringify({
  id: first.id, doc: first.doc, input: first.input, expect: first.expect,
}))

// helper sanity
console.log('equal(42,42):', equal(42, 42))
console.log('equal(null,undefined):', equal(null, undefined))
console.log('equalStrict(null,undefined):', equalStrict(null, undefined))
console.log('errorMatches(any):', errorMatches({ any: true, text: null, regex: false }, 'boom'))
console.log('errorMatches(substr):', errorMatches({ any: false, text: 'not found', regex: false }, 'Key NOT FOUND here'))
console.log('structMatch ok:', structMatch({ a: { b: 2 } }, { a: { b: 2 }, c: 9 }).ok)
console.log('structMatch fail:', JSON.stringify(structMatch({ a: { b: 2 } }, { a: { b: 3 } })))

File: test/proto/dart/smoke.dart
// Smoke check (not a test): prove the provider loads and normalizes, and that
// the summary numbers match the canonical TS reference.
//
// Run from the repo root once a Dart toolchain is available:
//   dart run test/proto/dart/smoke.dart
// or with an explicit corpus path:
//   dart run test/proto/dart/smoke.dart build/test/test.json

import 'provider.dart';

void main(List<String> args) {
  final p = args.isNotEmpty ? TestProvider.load(args[0]) : TestProvider.load();

  final fns = p.functions();
  print('functions: ${fns.join(', ')}');

  var total = 0;
  final expectKinds = <String, int>{};
  final inputKinds = <String, int>{};
  for (final fn in fns) {
    for (final e in p.entries(fn)) {
      total++;
      expectKinds[e.expect.kindName] = (expectKinds[e.expect.kindName] ?? 0) + 1;
      inputKinds[e.input.kindName] = (inputKinds[e.input.kindName] ?? 0) + 1;
    }
  }

  print('total entries: $total');
  print('expect kinds: value=${expectKinds['value'] ?? 0}, '
      'absent=${expectKinds['absent'] ?? 0}, '
      'match=${expectKinds['match'] ?? 0}, '
      'error=${expectKinds['error'] ?? 0}');
  print('input kinds: in=${inputKinds['in'] ?? 0}, '
      'args=${inputKinds['args'] ?? 0}, '
      'ctx=${inputKinds['ctx'] ?? 0}');

  final gp = p.entries('getpath', 'basic');
  if (gp.isNotEmpty) {
    final e = gp[0];
    print('getpath/basic[0]: id=${e.id}, doc=${e.doc}, '
        'input.kind=${e.input.kindName}, expect.kind=${e.expect.kindName}, '
        'expect.value=${e.expect.value}');
  }

  // helper sanity
  print('equal(42,42): ${equal(42, 42)}');
  print('equal(null,null): ${equal(null, null)}');
  print('errorMatches(any): '
      '${errorMatches(const ErrorCheck(any: true), 'boom')}');
  print('errorMatches(substr): '
      '${errorMatches(const ErrorCheck(any: false, text: 'not found'), 'Key NOT FOUND here')}');
  print('structMatch ok: '
      '${structMatch({'a': {'b': 2}}, {'a': {'b': 2}, 'c': 9}).ok}');
  final sm = structMatch({'a': {'b': 2}}, {'a': {'b': 3}});
  print('structMatch fail: ok=${sm.ok}, path=${sm.path}, '
      'expected=${sm.expected}, actual=${sm.actual}');
}
